src/lib/structures/game/BinbGame.ts

```diff
@@ -150,17 +150,20 @@ export class BinbGame extends Game {
 		return songsListenedTo * (1 + 1.5 / this.players.size);
 	}
 
-	protected getPlayers(voiceChannelMembers: Collection<string, GuildMember>) {
-		// TODO: Prompt players to join the game with a button.
-		const basePlayer: Omit<Player, 'id'> = { lastGameEntryTime: Date.now(), totalPlayTime: 0, songsListenedTo: 0 };
-		return new Collection(voiceChannelMembers.map<[Snowflake, Player]>((member) => [member.id, { ...basePlayer, id: member.id }]));
+	protected async getPlayers(voiceChannelMembers: Collection<string, GuildMember>) {
+		const joinButton = new MessageActionRow() //
+			.addComponents(
+				new MessageButton() //
+					.setCustomId(CustomIds.Join)
+					.setLabel('Join Game')
+					.setStyle(Constants.MessageButtonStyles.SUCCESS)
+			);
+
+		const embed = createEmbed(`Click the button to join this round of ${this.gameType} music trivia!`, EmbedColor.Primary);
+		await Promise.all(voiceChannelMembers.map((member) => member.send({ embeds: [embed], components: [joinButton] })));
```

Comment:
So what I'd do is something like:
(When sending messages directly to users, always make sure to catch in case they don't accept dms or some other reason)

```suggestion
		const promises = voiceChannelMembers.map((member) => member.send({ embeds: [embed], components: [joinButton] }).catch(() => null));
		const messages = await Promise.all(promises);

		// This is a promisified version of the global setTimeout, import it from `node:timers/promises`.
		// Since we await it, it will block the `getPlayers` function for 10 seconds too.
		await setTimeout(Time.Second* 10);

		const alsoPromisesIdk = messages
			// Filter out those who have been accepted or those who the bot failed to dm.
  			.filter((message) => message && !this.players.has(message.author.id))
  			.map((message) =>
    			message.edit({
      				embeds: [/* Too late! or smth */],
      				components: [],
   				})
  		);

		await Promise.all(alsoPromisesIdk);
```

For components, instead of removing them, you might wanna just disable them, idk